tests/functional/conftest.py

```diff
@@ -21,17 +22,29 @@ def fixture_dir(test_dir):
 def reset_gitlab(gl):
     # previously tools/reset_gitlab.py
     for project in gl.projects.list():
+        logging.info(f"Mark for deletion project: {project.name!r}")
         for deploy_token in project.deploytokens.list():
+            logging.info(
+                f"Mark for deletion token: {deploy_token.name!r} in "
+                f"project: {project.name!r}"
+            )
             deploy_token.delete()
         project.delete()
     for group in gl.groups.list():
+        logging.info(f"Mark for deletion group: {group.name!r}")
         for deploy_token in group.deploytokens.list():
+            logging.info(
+                f"Mark for deletion token: {deploy_token.name!r} in "
+                f"group: {group.name!r}"
+            )
             deploy_token.delete()
         group.delete()
     for variable in gl.variables.list():
+        logging.info(f"Mark for deletion variable: {variable.name!r}")
         variable.delete()
     for user in gl.users.list():
         if user.username != "root":
+            logging.info(f"Mark for deletion user: {user.username!r}")
```

Comment:
Just a suggestion for unique identifiers (also don't think variables have a `name` attribute) and style:

```suggestion
        logging.info(f"Marking for deletion project: {project.path_with_namespace!r}")
        for deploy_token in project.deploytokens.list():
            logging.info(
                f"Marking for deletion token: {deploy_token.username!r} in "
                f"project: {project.path_with_namespace!r}"
            )
            deploy_token.delete()
        project.delete()
    for group in gl.groups.list():
        logging.info(f"Mark for deletion group: {group.path_with_namespace!r}")
        for deploy_token in group.deploytokens.list():
            logging.info(
                f"Marking for deletion token: {deploy_token.username!r} in "
                f"group: {group.path_with_namespace!r}"
            )
            deploy_token.delete()
        group.delete()
    for variable in gl.variables.list():
        logging.info(f"Marking for deletion variable: {variable.key!r}")
        variable.delete()
    for user in gl.users.list():
        if user.username != "root":
            logging.info(f"Marking for deletion user: {user.username!r}")
```

Reply:
Done

Reply:
I guess 'groups' don't have `path_with_namespace` but they do have a `full_path` attribute. Using that.